### PR TITLE
SYS-3828: parachain-staking

### PR DESCRIPTION
**Pull request by ivan-cholakov**
Updated parachain-staking

### CODE REVIEW

File: pallets/parachain-staking/src/lib.rs [hunk 1]
```diff
@@ -2491,9 +2490,9 @@ pub mod pallet {
             );
         }
 
-        fn note_uncle(_author: T::AccountId, _age: T::BlockNumber) {
-            //TODO: can we ignore this?
-        }
+        // fn note_uncle(_author: T::AccountId, _age: BlockNumberFor<T>) {
+        //     //TODO: can we ignore this?
+        // }
```
Comment (thadouk) [inline pallets/parachain-staking/src/lib.rs:2495]: is this gone?
If so, can we remove?

File: pallets/parachain-staking/src/set.rs [hunk 2]
```diff
@@ -25,7 +25,18 @@ use sp_std::prelude::*;
 
 /// An ordered set backed by `Vec`
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Default, Clone, TypeInfo)]
+#[derive(
+    // Serialize,
+    // Deserialize,
```
Comment (thadouk) [inline pallets/parachain-staking/src/set.rs:30]:
```suggestion
```

File: pallets/parachain-staking/src/set.rs [hunk 3]
```diff
@@ -91,7 +102,18 @@ impl<T: Ord> From<Vec<T>> for OrderedSet<T> {
 
 /// An ordered set backed by `BoundedVec`
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo, Clone, MaxEncodedLen)]
+#[derive(
+    // Serialize,
+    // Deserialize,
```
Comment (thadouk) [inline pallets/parachain-staking/src/set.rs:107]:
```suggestion
```

**Review by thadouk — APPROVED** [review]

File: pallets/parachain-staking/Cargo.toml [hunk 4]
```diff
@@ -57,13 +62,16 @@ std = [
 	"serde",
 	"sp-runtime/std",
 	"sp-std/std",
+    "sp-io/std",
```
Comment (nahuseyoum) [inline pallets/parachain-staking/Cargo.toml:65]:
```suggestion
	"sp-io/std",
```

File: pallets/parachain-staking/src/lib.rs [hunk 5]
```diff
@@ -689,7 +688,7 @@ pub mod pallet {
         pub min_total_nominator_stake: BalanceOf<T>,
     }
 
-    #[cfg(feature = "std")]
+    // #[cfg(feature = "std")]
```
Comment (nahuseyoum) [inline pallets/parachain-staking/src/lib.rs:691]: Do we need this?

**Review by nahuseyoum — APPROVED** [review]

**Pull request merged**